packages/db/lib/gen-migration.mjs

```diff
@@ -0,0 +1,72 @@
+import fs from 'fs'
+import { stat } from 'fs/promises'
+import path from 'path'
+import { MigrateError } from './errors.mjs'
+import loadConfig from './load-config.mjs'
+
+async function generateMigration (_args) {
+  const { configManager } = await loadConfig({
+    string: ['to'],
+    boolean: ['rollback'],
+    alias: {
+      t: 'to',
+      r: 'rollback'
+    }
+  }, _args)
+  const migrationsDir = configManager.current.migrations.dir
+
+  // Check migrations directory exists
+  await checkMigrationsDirectoryExists(migrationsDir)
+
+  try {
+    await configManager.parseAndValidate()
+    const files = await fs.promises.readdir(migrationsDir)
+
+    const nextMigrationId = getNextMigrationId(files)
+
+    const nextDoMigrationName = `${nextMigrationId.toString().padStart(3, '0')}.do.sql`
+    await generateMigrationFile(migrationsDir, nextDoMigrationName)
+
+    const nextUndoMigrationName = `${nextMigrationId.toString().padStart(3, '0')}.undo.sql`
+    await generateMigrationFile(migrationsDir, nextUndoMigrationName)
+  } catch (err) {
+    console.error(err.message)
+    process.exit(1)
+  }
+}
+
+function getNextMigrationId (migrationFileName) {
+  if (migrationFileName.length === 0) {
+    return 1
+  }
+
+  const chunks = migrationFileName.pop().split('.')
```

Comment:
Please use fs basename and extname functions.

Reply:
Done